### ISSUE TITLE
ODT Invoice generation currency format

### ISSUE THREAD

**Issue by arunasr** [Dolibarr/dolibarr#5539]
# Bug

When substituting currency fields in ODT template, Dolibarr  seems to be using SQL internal number format (as returned by `price2num` function) with all zeroes stripped. 
The currency fields should be displayed in the preferred locale format, with fixed decimal places present even if zeroes. 
I understand that currency fields with `_locale` suffix were meant to provide this format, but these fields do not work in the 3.9 version (they are also not present in the invoice template provided with the program).
## Environment
- **Version**: 3.9.0
- **OS**: Debian 8.5 Jessie
- **Web server**: Apache 2.4.10
- **PHP**: 5.6.23
- **Database**: MySQL 5.5.49
- **URL(s)**: /compta/facture.php
  ## Expected and actual behavior
  Display prices in either my company's locale currency format, or in the target company locale format.
## Steps to reproduce the behavior

Generate invoice from ODT template

[FA1607-0002_Invoice.odt.zip](https://github.com/Dolibarr/dolibarr/files/381267/FA1607-0002_Invoice.odt.zip)

**Comment by eldy** [comment]
Do you experience pb with 4.0RC ?
If yes, can you give the name of the tag tha is concerned. Is it
object_total_ttc_locale ? with price 845.1 pound in your sample that should be 845.10 ?

**Comment by arunasr** [comment]
I did not try with 4.0RC.

 From the code it looks like {xyz_locale} are supposed to be substituted 
with the regional number, but when I tried using them, they do not get 
substituted. The {xyz} fields get substituted with the the DB cannonical 
number representation as returned by price2num(), without thousands 
separator and with all decimal zeroes stripped.

My current template is attached; it does not use {xyz_locale} fields 
because that does not work. The template that is delivered with the 
released software also does not contain any {_locale} fields.

On a different note, the line date fields also do not work (such as 
{line_date_start}). They get substituted with the milliseconds since 
epoch, not the date string. How can I get proper date substitution? Note 
that {line_date_start_rfc} also does not work.

Thanks
Arunas
